src/test/mir-opt/const_prop/issue-67019/rustc.main.ConstProp.diff

```diff
@@ -0,0 +1,47 @@
+- // MIR for `main` before ConstProp
++ // MIR for `main` after ConstProp
+  
+  fn main() -> () {
+      let mut _0: ();                      // return place in scope 0 at $DIR/issue-67019.rs:10:11: 10:11
+      let _1: ();                          // in scope 0 at $DIR/issue-67019.rs:11:5: 11:20
+      let mut _2: ((u8, u8),);             // in scope 0 at $DIR/issue-67019.rs:11:10: 11:19
+      let mut _3: (u8, u8);                // in scope 0 at $DIR/issue-67019.rs:11:11: 11:17
+  
+      bb0: {
+          StorageLive(_1);                 // bb0[0]: scope 0 at $DIR/issue-67019.rs:11:5: 11:20
+          StorageLive(_2);                 // bb0[1]: scope 0 at $DIR/issue-67019.rs:11:10: 11:19
+          StorageLive(_3);                 // bb0[2]: scope 0 at $DIR/issue-67019.rs:11:11: 11:17
+          _3 = (const 1u8, const 2u8);     // bb0[3]: scope 0 at $DIR/issue-67019.rs:11:11: 11:17
+                                           // ty::Const
+                                           // + ty: u8
+                                           // + val: Value(Scalar(0x01))
+                                           // mir::Constant
+-                                          // + span: $DIR/issue-67019.rs:11:12: 11:13
++                                          // + span: $DIR/issue-67019.rs:11:11: 11:17
+                                           // + literal: Const { ty: u8, val: Value(Scalar(0x01)) }
+                                           // ty::Const
+                                           // + ty: u8
+                                           // + val: Value(Scalar(0x02))
+                                           // mir::Constant
+-                                          // + span: $DIR/issue-67019.rs:11:15: 11:16
++                                          // + span: $DIR/issue-67019.rs:11:11: 11:17
```

Comment:
@wesleywiser @oli-obk What is happening here, why is the span of the `mir::Constant` changing?

Reply:
I suspect [`ConstPropagator::replace_with_const()`](https://github.com/rust-lang/rust/blob/master/src/librustc_mir/transform/const_prop.rs#L638) is getting called and is using the `Span` from the MIR `Statement` which creates a new `Const` and replaces the existing one with the correct `Span`. We probably just need to early return out of `replace_with_const()` if `rvalue` is already a `Const`.